File: week2_algorithic_warm-up/lcm.py
def lcm_naive(a, b):
    for l in range(1, a*b+1):
        if l % a == 0 and l % b == 0:
            return l

    return a*b

def gcd(a, b):
    if (b == 0):
        return a

    a_prime = a%b
    return gcd(b, a_prime)

def lcm(a, b):
    return (a*b)/gcd(a,b)

a = int(input())
b = int(input())

print(lcm(a, b))

File: week2_algorithic_warm-up/arithmetic/lcm/lcm.cpp
#include <iostream>

long long gcd(long long int, long long int);
long long lcm(long long, long long);

int main()
{
    long long int a, b;
    std::cin >> a >> b;
    std::cout << lcm(a, b) << std::endl;
    return 0;
}

long long gcd(long long int a, long long int b) {
    if (b == 0)
        return a;

    int a_prime = a % b;
    return gcd(b, a_prime);
}

long long lcm(long long a, long long b) {
    return ((long long) (a*b)/gcd(a, b));
}
